partiql-spi/src/main/kotlin/org/partiql/spi/function/builtins/internal/Accumulator.kt

```diff
@@ -97,12 +97,16 @@ private fun Long.checkOverflowPlus(other: Long): Number {
 }
 
 internal fun checkIsBooleanType(funcName: String, value: Datum) {
+    if (value.type.code() == PType.VARIANT) {
```

Comment:
Is there a reason we add this variant type check for booleans but not for numbers in `checkIsNumberType`?